cql3/statements/select_statement.cc

```diff
@@ -2032,7 +2032,10 @@ std::unique_ptr<prepared_statement> select_statement::prepare(data_dictionary::d
             && !restrictions->need_filtering()  // No filtering
             && group_by_cell_indices->empty()   // No GROUP BY
             && db.get_config().enable_parallelized_aggregation()
-            && !is_local_table();
+            && !is_local_table()
+            && !( // Do not parallelize the request if it's single partition read
+                restrictions->partition_key_restrictions_is_all_eq() 
+                && restrictions->partition_key_restrictions_size() == 1);
```

Comment:
I have some doubts about this logic:
1. What if we have several partition-key components, e.g., p1 and p2, and `WHERE p1=2 AND p2=3`? Won't this have partition_key_restrictions_size() > 1 and fail this test?
2. What if we have not a single partition read, but a single `WHERE token(p) = 123456`, will it avoid the parallelization too (it can).

Whatever way you plan to test this fix, please check these cases as well.

Reply:
Yes, both cases are not parallelized. Added them to the test

Reply:
The 1st example wasn't handled. Fixed it and included both cases in test